Error: reference to member 'unavailable' cannot be resolved without a contextual type

## System info

Issue occurs on: iOS
Plugin name: connectivity_plus
Plugin version: 2.1.0
Flutter doctor output:

```
[✓] Flutter (Channel stable, 2.8.0, on macOS 12.0.1 21A559 darwin-x64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.2)
[✓] Chrome - develop for the web
[!] Android Studio
    ✗ Unable to find bundled Java version.
[✓] Android Studio (version 2020.3)
[✓] Android Studio (version 2020.3)
[✓] Connected device (3 available)

! Doctor found issues in 1 category.
```

## Steps to Reproduce
1. Just trying to build and run the app
2. Error occurred

## Logs
```
/usr/local/Caskroom/flutter/2.2.3/flutter/.pub-cache/hosted/pub.dartlang.org/connectivity_plus-2.1.0/ios/Classes/ReachabilityConnectivityProvider.swift:9:41: error: reference to member 'unavailable' cannot be resolved without a contextual type
        switch reachability?.connection ?? .unavailable {
                                            ^
```

```
[✓] Flutter (Channel stable, 2.8.0, on macOS 12.0.1 21A559 darwin-x64, locale en-GB)
    • Flutter version 2.8.0 at /usr/local/Caskroom/flutter/2.2.3/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cf44000065 (8 days ago), 2021-12-08 14:06:50 -0800
    • Engine revision 40a99c5951
    • Dart version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/kongtengyee/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/kongtengyee/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/kongtengyee/Library/Android/sdk
    • Java binary at: /Users/kongtengyee/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-1/203.7784292/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/kongtengyee/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-1/203.7784292/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/kongtengyee/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-1/203.7678000/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Connected device (3 available)
    • Android SDK built for x86 64 (mobile) • emulator-5556                        • android-x64    • Android 10 (API 29) (emulator)
    • iPhone 13 (mobile)                    • CBA6DB57-987C-4A4A-992E-8AAAF9755C16 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • Chrome (web)                          • chrome                               • web-javascript • Google Chrome 96.0.4664.110

! Doctor found issues in 1 category.
```

--- a/packages/connectivity_plus/connectivity_plus/CHANGELOG.md
+++ b/packages/connectivity_plus/connectivity_plus/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 2.2.2
+
+- Reachability.swift ".unavailable" for iOS is deprecated.
+
 ## 2.2.1
 
 - Bump `nm` plugin to 0.5.0 (connectivity_plus_linux)

--- a/packages/connectivity_plus/connectivity_plus/ios/Classes/ReachabilityConnectivityProvider.swift
+++ b/packages/connectivity_plus/connectivity_plus/ios/Classes/ReachabilityConnectivityProvider.swift
@@ -6,7 +6,7 @@ public class ReachabilityConnectivityProvider: NSObject, ConnectivityProvider {
 
   public var currentConnectivityType: ConnectivityType {
     let reachability = try? self.reachability ?? Reachability()
-    switch reachability?.connection ?? .unavailable {
+    switch reachability?.connection {
     case .wifi:
       return .wifi
     case .cellular:

--- a/packages/connectivity_plus/connectivity_plus/pubspec.yaml
+++ b/packages/connectivity_plus/connectivity_plus/pubspec.yaml
@@ -1,6 +1,6 @@
 name: connectivity_plus
 description: Flutter plugin for discovering the state of the network (WiFi & mobile/cellular) connectivity on Android and iOS.
-version: 2.2.1
+version: 2.2.2
 homepage: https://plus.fluttercommunity.dev/
 repository: https://github.com/fluttercommunity/plus_plugins/tree/main/packages/
 

